extensions-utils/src/dfx.rs

```diff
@@ -50,6 +54,57 @@ where
         })
 }
 
+/// Calls a binary that was delivered with an extension tarball.
+///
+/// # Returns
+/// - On success, returns stdout as a string.
+/// - On error, returns an error message including stdout and stderr.
+#[context("Calling {} CLI failed, or, it returned an error.", binary_name)]
+pub fn call_extension_bundled_binary<S, I>(binary_name: &str, args: I) -> anyhow::Result<String>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let extension_binary_path =
+        std::env::current_exe().map_err(|e| anyhow::anyhow!("Failed to get current exe: {}", e))?;
+    let extension_dir_path = extension_binary_path
+        .parent()
+        .unwrap_or_else(|| Path::new("."));
+    let binary_to_call = extension_dir_path.join(binary_name);
+
+    let mut command = Command::new(&binary_to_call);
+    command.args(args);
+    // The sns command line tool itself calls dfx; it should call this dfx.
+    // The sns command line tool should not rely on commands not packaged with dfx.
+    // The same applies to other bundled binaries.
+    command.env(
```

Comment:
where is the path to dfx coming from?  Some of the use cases for dfx extensions have the extension call dfx one or more times.  It used to be the case, and may still be, that `sns-cli`, `ic-admin`, and/or `ic-nns-init` called dfx.  For this reason we talked about passing the path to dfx to the extension when calling it.   (It's understood that the experience might not be great if this means asking for an identity passcode more than once).

Reply:
right, forgot about it